src/mavlink-router/main.cpp

```diff
@@ -224,19 +224,24 @@ static int add_endpoint_address(const char *name, size_t name_len, const char *i
     }
 
     if (ip) {
-        free(conf->address);
         conf->address = strdup(ip);
         if (!conf->address) {
             ret = -ENOMEM;
             goto fail;
         }
-    }
-
-    if (!conf->address) {
+    } else {
```

Comment:
all these changes in this function seem good, but should be in a separate commit as a cleanup. I think this function got reworked at some time and these checks remained, but don't make sense anymore.

Reply:
In this one, btw: `ip` is mandatory... so just remove the `if (ip)` and let the check for ENOMEM. It would be good to put a `assert_or_return(ip != nullptr, -EINVAL)` in the beginning of the function.

Reply:
Reverted - will create a new cleanup PR.